src/plots/stackedAreaPlot.ts

```diff
@@ -202,6 +181,33 @@ export module Plots {
       return { x: pixelPoint.x, y: scaledYValue };
     }
 
+    protected _getDataToDraw() {
+      if (!this.downsamplingEnabled()) {
+        return super._getDataToDraw();
+      }
+      let dataToDraw = new Utils.Map<Dataset, any[]> ();
+      if (this.datasets().length === 0) {
+        return dataToDraw;
+      }
+      let overallFilteredDataIndices: number[] = [];
+      this.datasets().forEach((dataset, i) => {
+        let data = dataset.data();
+        let filteredDataIndices = data.map((d, i) => i);
+        if (this.croppedRenderingEnabled()) {
+          filteredDataIndices = this._filterCroppedRendering(dataset, filteredDataIndices);
+        }
+        if (this.downsamplingEnabled()) {
+          filteredDataIndices = this._filterDownsampling(dataset, filteredDataIndices);
+        }
+        overallFilteredDataIndices = overallFilteredDataIndices.concat(filteredDataIndices
+          .filter((index) => {return overallFilteredDataIndices.indexOf(index) < 0; }));
+      });
+      this.datasets().forEach((dataset, i) => {
+        let data = dataset.data();
+        dataToDraw.set(dataset, [overallFilteredDataIndices.sort((a, b) => a - b).map((d, i) => data[d])]);
```

Comment:
Remove the `i` parameter in the map.  Also replace the `d` parameter there.  It is incredibly missing as an `index` is actually being passed as the first argument.